app/ts/ManualInput/CellValidityService/CellValidityService.ts

```diff
@@ -153,3 +157,106 @@ function isValidCell(cell: Effect | Distribution, unit: IUnitOfMeasurement) {
       );
   }
 }
+
+export function getOutOfBoundsError(
+  datasourceId: string,
+  effects: Record<string, Record<string, Effect>>,
+  distributions: Record<string, Record<string, Distribution>>,
+  inputUpperBound: number
+): string {
+  const upperBound = inputUpperBound;
+  const lowerBound = 0;
+  if (
+    hasOutOfDoundsEffect(effects[datasourceId], lowerBound, upperBound) ||
+    hasOutOfDoundsDistribution(
+      distributions[datasourceId],
+      lowerBound,
+      upperBound
+    )
+  ) {
+    return `Some cell values are out of bounds [0, ${upperBound}]`;
+  } else {
+    return '';
+  }
+}
+
+function hasOutOfDoundsEffect(
+  effects: Record<string, Effect>,
+  lowerBound: number,
+  upperBound: number
+): boolean {
+  return _.some(_.map(effects), (effect: Effect) => {
+    if (isValue(effect)) {
+      return isValueOutOfDounds(effect, lowerBound, upperBound);
+    } else if (isValueCI(effect) || isRange(effect)) {
+      return areBoundsInvalid(effect, lowerBound, upperBound);
+    } else {
+      return false;
+    }
+  });
+}
+
+function hasOutOfDoundsDistribution(
+  distributions: Record<string, Distribution>,
+  lowerBound: number,
+  upperBound: number
+): boolean {
+  return _.some(_.map(distributions), (distribution: Distribution) => {
+    if (isValue(distribution)) {
+      return isValueOutOfDounds(distribution, lowerBound, upperBound);
+    } else if (isRange(distribution)) {
+      return areBoundsInvalid(distribution, lowerBound, upperBound);
+    } else if (isNormalDistribution(distribution)) {
+      return areParametersOutOfBound(distribution, lowerBound, upperBound);
+    } else {
+      return false;
```

Comment:
A switch over type might be nicer to have here instead of ifs and helper functions.